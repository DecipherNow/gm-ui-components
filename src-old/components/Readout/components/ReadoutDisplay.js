import styled from "styled-components";
import { PropTypes } from "prop-types";

import {
  BORDER_RADIUS_BASE,
  FONT_STACK_BASE,
  COLOR_CONTENT
} from "style/styleVariables";
import { spacingScale } from "style/styleFunctions";

const ReadoutDisplay = styled.div`
  color: ${COLOR_CONTENT};
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
  align-items: stretch;
  justify-content: center;
  border-radius: ${BORDER_RADIUS_BASE};
  margin: 0 ${spacingScale(1)} ${spacingScale(1)} 0;
  position: relative;
  overflow: hidden;
  font-family: ${FONT_STACK_BASE};

  &:before {
    content: "";
    opacity: 0.2;
    background-color: ${props => props.color};
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 0;
  }

  &:after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    border-bottom: 3px solid ${props => props.color};
  }

  &:first-child:last-child {
    flex-grow: 0;
  }

  ${props =>
    props.primary &&
    `
    &:before {
      opacity: 0.3;
    }
  `};
`;

ReadoutDisplay.propTypes = {
  color: PropTypes.string,
  primary: PropTypes.bool
};

export default ReadoutDisplay;
