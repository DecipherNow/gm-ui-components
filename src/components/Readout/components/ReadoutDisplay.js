import styled from "styled-components";
import { PropTypes } from "prop-types";

import { spacingScale } from "style/styleFunctions";
import { keen } from "style/theme";

const ReadoutDisplay = styled.div`
  color: ${({ theme }) => theme.COLOR_CONTENT_DEFAULT};
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
  align-items: stretch;
  justify-content: center;
  border-radius: ${({ theme }) => theme.CORNER_RADIUS_CARD_DEFAULT};
  margin: 0 ${spacingScale(1)} ${spacingScale(1)} 0;
  position: relative;
  overflow: hidden;
  font-family: ${({ theme }) => theme.FONT_STACK_DEFAULT};
  box-shadow: inset 0 0 0 1px ${({ theme }) => theme.COLOR_KEYLINE_DEFAULT},
    ${props => props.color && "inset 0 -43px 0 -40px " + props.color};
  z-index: 1;

  &:before {
    content: "";
    opacity: 0.2;
    background-color: ${props => props.color};
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 0;
  }

  &:first-child:last-child {
    flex-grow: 0;
  }

  ${props =>
    props.primary &&
    `
    &:before {
      opacity: 0.3;
    }
  `};
`;

ReadoutDisplay.propTypes = {
  color: PropTypes.string,
  primary: PropTypes.bool
};

ReadoutDisplay.defaultProps = {
  color: "transparent",
  theme: keen
};

export default ReadoutDisplay;
