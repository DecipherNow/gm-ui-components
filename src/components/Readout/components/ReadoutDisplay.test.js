import React from "react";
import { shallow } from "enzyme";
import ReadoutDisplay from "./ReadoutDisplay";

describe("ReadoutDisplay", () => {
  it("matches snapshot when primary is true", () => {
    const aReadoutDisplay = shallow(
      <ReadoutDisplay primary={true} color={"#000"} />
    );
    expect(aReadoutDisplay).toMatchSnapshot();
  });
  it("matches snapshot when primary is false", () => {
    const aReadoutDisplay = shallow(
      <ReadoutDisplay primary={false} color={"#000"} />
    );
    expect(aReadoutDisplay).toMatchSnapshot();
  });
});
