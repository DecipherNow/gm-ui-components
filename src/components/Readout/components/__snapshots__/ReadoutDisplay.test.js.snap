// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadoutDisplay matches snapshot when primary is false 1`] = `
.c0 {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 3px;
  margin: 0 8px 8px 0;
  position: relative;
  overflow: hidden;
}

.c0:before {
  content: "";
  opacity: 0.2;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
}

.c0:after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-bottom: 3px solid;
}

.c0:first-child:last-child {
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
}

<div
  className="c0"
/>
`;

exports[`ReadoutDisplay matches snapshot when primary is true 1`] = `
.c0 {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 3px;
  margin: 0 8px 8px 0;
  position: relative;
  overflow: hidden;
}

.c0:before {
  content: "";
  opacity: 0.2;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
}

.c0:after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-bottom: 3px solid;
}

.c0:first-child:last-child {
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
}

.c0:before {
  opacity: 0.3;
}

<div
  className="c0"
/>
`;
